### ISSUE TITLE
Examples of usage with non-coroutine based approaches

### ISSUE THREAD

**Issue by kavalerov** [ably/ably-asset-tracking-android#309]
Right now the SDK is heavily based on Kotlin coroutines. This is how it works internally, and it also is exposed in our APIs. 

However, not all users will be using coroutines in their apps. They may use alternative approaches (like [Rx](https://github.com/ReactiveX/RxKotlin)), or not use any such approach at all. 

# Should we provide examples of SDK usage from code written using Rx, or just "standard" Android code without coroutines?
# Should we change/add APIs that do not impose requirements and/or additional dependencies on users in terms of our coroutines usage?

**Comment by QuintinWillison** [comment]
I agree that we should illustrate as many mechanisms for using the SDK as possible, however bear in mind that [the Android documentation](https://developer.android.com/kotlin/coroutines) specifically states:

> Coroutines is our recommended solution for asynchronous programming on Android.
